### ISSUE TITLE
Using an IValueValidator implementation validating a List<Integer> parameter, new instance created receiving list of one value for each value in final list

### ISSUE THREAD

**Issue by devynms** [cbeust/jcommander#322]
## Issue ##
When using a parameter that is a `List<Integer>` maked with `variableArity=true` and `validateValueWith` set to a custom validator implementing `IValueValidator<List<Integer>>`, I only see the elements of my parameter in lists of size 1 one by one, and I am unable to validate properties of the whole list. Unfortunately, http://jcommander.org/ doesn't have anything about `IValueValidator`, and its [API docs](http://jcommander.org/apidocs/com/beust/jcommander/IValueValidator.html) don't mention what the behavior should be. However, it seems reasonable to assume that the intended behavior would be to receive the whole list in the `validate` method.

I am using Java 8 and JCommander 1.7.

## Replicating the issue ##
I currently have the following class, `Arguments` which I parse using `new JCommander(myArgs, commandLineArgs)`. `Arguments` has the relevant following code:

```
public class Arguments {
  @Parameter(names = {"-v", "--values"},
            required = true,
            variableArity = true,
            validateValueWith = MyValidator.class)
    private List<Integer> values;

    public int[] getValues() {
        return this.values.stream().mapToInt(Integer::intValue).toArray();
    }

...

public static class MyValidator implements IValueValidator<List<Integer>> {
        @Override
        public void validate(String name, List<Integer> values) throws ParameterException {
            ... BODY ...
        }
    }
}
```
Currently, I want to write a `validate` method that confirms that the list `values` is a list of strictly increasing integers. I am running in with the following command line arguments which I intend to cause a failure: `-v 1 4 2`.

Currently, when I print out the `List<Integer> values` object in the `validate` body, I see the following lists:
{ 1 }
{ 4 }
{ 2 }
Where I expected to see the following:
{ 1, 4, 2 }

I have the following JUnit tests which is currently passing and reinforces my expectations:
```
    @Test
    public void acceptsSpaceSeparatedYearBuckets() {
        String[] commandLineArgs = "-v 1 3 4".split("\\s+");
        int[] expectedValues = {1, 3, 4};
        Arguments myArgs = new Arguments();
        new JCommander(myArgs, commandLineArgs);
        int[] actualValues = myArgs.getValues();
        assertArrayEquals(expectedBuckets, actualBuckets);
    }
```

## Failed workarounds ##

- **Setting a private field to contain the last value seen in the array**
I tried to perform my validation by remembering the last value seen in the array, starting with the minimum integer value. However, it appears that a new instance of my validator was created for every element, as setting the previous value did not appear to affect other validations.

- **Implementing `IValueValidator<Integer>` instead.**
I thought maybe that the intention of the validator was to validate every value of the array individually, and that validating the whole array was not intended. However, when doing so, I received the following exception:
`Exception in thread "main" java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.Integer`

**Comment by devynms** [comment]
I had the same issue using version 1.35 of JCommander.

It looks like Issue #269 may be related, but I'm not sure.

**Comment by cbeust** [comment]
Could you try with the latest version, just in case?  It's 1.61.

If it still fails, could you please attach a failing test? I tried to turn your code above in one but I'm missing a few parts.

**Comment by devynms** [comment]
cbeust, thanks for taking the time to respond.

My code was previously using version 1.35, I changed it to 1.7 mistakenly thinking it was more up to date. I have changed to 1.61 and ran my changes with a test failing on what I currently believe to be an issue. I've copied the relevant code (the names are different than they are in my code base, so I apologize if it doesn't run as-is).

**Arguments.java**
```
public class Arguments {

    @Parameter(names = {"-v", "--values"},
            required = true,
            variableArity = true,
            validateValueWith = ValuesValidator.class)
    private List<Integer> values;

    public static class ValuesValidator implements IValueValidator<List<Integer>> {

        @Override
        public void validate(String name, List<Integer> values) throws ParameterException {
            int previous = Integer.MIN_VALUE;
            for (int i : values) {
                if (i <= previous) {
                    throw new ParameterException("Invalid: values should be stricly increasing.");
                }
                previous = i;
            }
        }
    }
}
```

**ArgumentsTest.java**
```
public class ArgumentsTest {

    @Rule
    public final ExpectedException exception = ExpectedException.none();

    @Test
    public void rejectsNonIncreasingValues() {
        exception.expect(ParameterException.class);
        String[] commands = "-v 1 5 2".split("\\s+");
        new JCommander(new CountGroupedByYearArguments(), commands);
    }
}
```

**Test output**
```
ArgumentsTest > rejectsNonIncreasingYearBuckets FAILED
    java.lang.AssertionError: Expected test to throw an instance of com.beust.jcommander.ParameterException
FAILURE after 0.049 sec: 3 Passed, 1 Failed, 0 Skipped
```

Running my project directly, I tried just doing `System.out.println(Arrays.toString(values.toArray()));` at the top of the validation method to see what I'd get, and I got (with "-v 1 2 4" on the command line) `[1]`, `[2]`, and `[4]`, where my intuition says I should see `[1, 2, 4]`.

**Comment by cbeust** [comment]
Should be fixed in 1.62.
